backend/routes/canvas.go

```diff
@@ -14,22 +14,29 @@ func InitCanvasRoutes() {
 }
 
 func initCanvas(w http.ResponseWriter, r *http.Request) {
-	// TODO: Check if canvas already exists
-	totalBitSize := core.ArtPeaceBackend.CanvasConfig.Canvas.Width * core.ArtPeaceBackend.CanvasConfig.Canvas.Height * core.ArtPeaceBackend.CanvasConfig.ColorsBitWidth
-	totalByteSize := (totalBitSize / 8)
-	if totalBitSize%8 != 0 {
-		// Round up to nearest byte
-		totalByteSize += 1
+	// Check if the 'canvas' property exists in core.ArtPeaceBackend.CanvasConfig
+	if core.ArtPeaceBackend.CanvasConfig != nil && core.ArtPeaceBackend.CanvasConfig.Canvas != nil {
```

Comment:
This check is good, but I was thinking of having another check like this `core.ArtPeaceBackend.Databases.Redis.Exists(context.Background(), "canvas").Val() == 1` to make sure the canvas does not already exist in Redis's DB.

Reply:
Alright, thank you for the feedback.
Adding the check now